packages/@aws-cdk/aws-synthetics/lib/canary.ts

```diff
@@ -0,0 +1,255 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as s3 from '@aws-cdk/aws-s3';
+import * as cdk from '@aws-cdk/core';
+import { CanaryBase } from './canary-base';
+import { CfnCanary } from './synthetics.generated';
+
+/**
+ * Optional properties for a canary
+ */
+export interface CanaryOptions extends cdk.ResourceProps {
+  /**
+   * The s3 location that stores the data of the canary runs.
+   *
+   * @default - A new s3 bucket will be created.
+   */
+  readonly artifactBucket?: s3.IBucket;
+
+  /**
+   * Canary execution role.
+   *
+   * This is the role that will be assumed by the canary upon execution.
+   * It controls the permissions that the canary will have. The Role must
+   * be assumable by the 'lambda.amazonaws.com' service principal.
+   *
+   * The default Role automatically has permissions granted for Canary execution.
+   * If you provide a Role, you must add the relevant policies yourself.
+   *
+   * The relevant policies are "s3:PutObject", "s3:GetBucketLocation", "s3:ListAllMyBuckets",
+   * "cloudwatch:PutMetricData", "logs:CreateLogGroup", "logs:CreateLogStream", "logs:PutLogEvents".
+   *
+   * @default - A unique role will be generated for this canary.
+   * Both supplied and generated roles can always be changed by calling 'addToRolePolicy'.
+   */
+  readonly role?: iam.IRole;
+
+  /**
+   * How many seconds the canary should run before timing out.
+   *
+   * @default - the smaller value between frequency and 900 seconds
+   */
+  readonly timeout?: cdk.Duration;
+
+  /**
+   * The amount of memory, in MB, that is allocated to your Canary. Must be a multiple of 64 and inside the range 960 to 3008.
+   *
+   * @default - 960
+   */
+  readonly memorySize?: number;
```

Comment:
I reckon there is a `Size` class (similar to `Duration`) that woulds offer unit-safety.

Personal taste: if your measurement property is typed in a unit-agnostic way, then the unit **must** be in the property name. But it's always better to use a unit-aware type.